homeassistant/components/wallbox/helpers.py

```diff
@@ -0,0 +1,59 @@
+"""Helpers for Wallbox integration."""
+
+from datetime import datetime as datetime_sys
+
+from dateutil import tz
+
+import homeassistant.util.dt as dt_util
+
+
+def map_schedule_item_to_ha(item: dict) -> dict:
+    """Map local schedule item to be more usable with HA."""
+    converted = {}
+    converted["start"] = _convert_to_local_time(item["start"])
+    converted["stop"] = _convert_to_local_time(item["stop"])
+    converted["id"] = item["id"]
+    converted["enable"] = item["enable"]
+    converted["days"] = item["days"]
+    if "max_current" in item:
+        converted["max_current"] = item["max_current"]
+    if "max_energy" in item:
+        converted["max_energy"] = item["max_energy"]
+    converted["created_at"] = item["created_at"]
+    return converted
+
+
+def map_ha_item_to_schedule(item: dict) -> dict:
+    """Map HA schedule item for API."""
+    converted = {}
+    converted["start"] = _convert_to_utc_time_str(item["start"])
+    converted["stop"] = _convert_to_utc_time_str(item["stop"])
+    converted["id"] = item["id"]
+    converted["enable"] = item["enable"]
+    converted["days"] = item["days"]
+    if "max_current" in item:
+        converted["max_current"] = item["max_current"]
+    if "max_energy" in item:
+        converted["max_energy"] = item["max_energy"]
+    return converted
+
+
+def _convert_to_local_time(time: str) -> str:
+    time_obj = datetime_sys.strptime(time, "%H%M").time()
+    utc_time = dt_util.now().replace(
```

Comment:
i think you can skip this local time var, I think you are creating 2 idential datetime objects here. Suggestion:
time_obj = datetime_sys.strptime(time, "%H:%M")
return dt_util.as_local(time_obj).strftime("%H%M")

And then just move it to the main function

Reply:
Went as that first, but bumped into an issue:
The time_obj received through strptime using hours and minutes only, the date will be set as '1900-01-01'

Using dt_util.as_local for 1900-01-01 date (including time) will not do the conversion to local timezone properly. 
So I decided to make a new object from now() and just replace the hours and minutes. That solved the problem.

So, most probably dt_util.as_local have some restrictions about a valid date or alternatively a bug in implementation.

Ps. I think in your sample the timezone should be overridden as UTC though:
  time_obj = datetime_sys.strptime(wallbox_time_str, "%H%M").replace(
      tzinfo=tz.UTC
  )

Reply:
ok, something like this: datetime.now().replace(hour=int(item['stop'][0:2]), minute=int(item['stop'][2:4]), second=0, microsecond=0)

Reply:
Pushed the changes requested.

Still decided to keep the time conversion on their own function in order to keep the code more readable. 
Though the conversion code(s) are optimised, moved to the coordinator and helper deleted.